exec/java-exec/src/main/java/org/apache/drill/exec/store/dfs/DrillFileSystem.java

```diff
@@ -788,51 +785,18 @@ public void removeXAttr(final Path path, final String name) throws IOException {
 
   /**
    * Returns an InputStream from a Hadoop path. If the data is compressed, this method will return a compressed
-   * InputStream depending on the codec.  Note that if the results of this method are sent to a third party parser
-   * that works with bytes or individual characters directly, you should use the openDecompressedInputStream method.
+   * InputStream depending on the codec.
    * @param path Input file path
    * @return InputStream of opened file path
    * @throws IOException If the file is unreachable, unavailable or otherwise unreadable
    */
   public InputStream openPossiblyCompressedStream(Path path) throws IOException {
-    CompressionCodec codec = codecFactory.getCodec(path); // infers from file ext.
+    CompressionCodec codec = getCodec(path); // infers from file ext.
+    InputStream inputStream = open(path);
```

Comment:
@vvysotskyi Hello. Is a better way to reduce mem usage when `codec` is not null?
```java
if (codec != null) {
  return codec.createInputStream(open(path));
} else {
  return open(path);
}
```
instead of
```java
CompressionCodec codec = getCodec(path);
InputStream inputStream = open(path);  // 1st
if (codec != null) {
  inputStream = codec.createInputStream(inputStream);  // 2nd
}
...
```

Reply:
Sorry, but I don't see how it reduces memory usage. When calling the `codec.createInputStream(open(path))`, Java will call the `open(path)` method, and after that will call the `createInputStream()` with its result, so these two versions of the code actually behave in the same way.

Reply:
@vvysotskyi 
This looks good.  Did you test it on s3 with a PCAP file to confirm that this works?

Reply:
Yes, I've checked it on s3 with several forms of the query (but with the same PCAP file).